pennylane/tape/qnode.py

```diff
@@ -112,7 +112,8 @@ class QNode:
         caching (int): Number of device executions to store in a cache to speed up subsequent
             executions. A value of ``0`` indicates that no caching will take place. Once filled,
             older elements of the cache are removed and replaced with the most recent device
-            executions to keep the cache up to date.
+            executions to keep the cache up to date. The cache is not available for
+            gradient-based calculations.

```

Comment:
Wouldn't one of the best uses of the cache be for gradient-based calculations?

Reply:
This would be fixed by [Tom's PR](https://github.com/PennyLaneAI/pennylane/pull/851) which moves caching to the device...